libbpf-rs/src/map.rs

```diff
@@ -211,6 +214,56 @@ impl Map {
         })
     }
 
+    /// Open a previously pinned map from it's path.
+    ///
+    /// # Panics
+    /// If the path contains null bytes.
+    pub fn from_pinned_path<P: AsRef<Path>>(path: P) -> Result<Self> {
+        let mut p = path.as_ref().as_os_str().as_bytes().to_vec();
+        if p.last() != Some(&0) {
+            p.push(0);
+        }
+        let p = CString::from_vec_with_nul(p).expect("path contained null bytes");
+        let fd = parse_ret_i32(unsafe {
+            // SAFETY
+            // p is never null since we allocated ourselves.
+            libbpf_sys::bpf_obj_get(p.as_ptr())
+        })?;
+        Self::from_fd(fd)
```

Comment:
Would you mind using the non-generic inner function pattern here to curb potential bloat? https://www.possiblerust.com/pattern/non-generic-inner-functions